subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyUnresolvedModuleIntegrationTest.groovy

```diff
@@ -0,0 +1,407 @@
+        """
+
+        when:
+        fails('resolve')
+
+        then:
+        assertDependencyMetaDataReadTimeout(moduleA)
+        !downloadedLibsDir.isDirectory()
+
+        where:
+        _ | protocol
+        _ | 'http'
+        _ | 'https'
+    }
+
+    @Unroll
+    def "fails concurrent application dependency resolution if #protocol connection exceeds timeout"() {
+        given:
+        MavenHttpModule moduleB = publishMavenModule(mavenHttpRepo, 'b')
+        MavenHttpModule moduleC = publishMavenModule(mavenHttpRepo, 'c')
+        blockingForProtocol(protocol, moduleA.pom, moduleB.pom, moduleC.pom)
+
+        buildFile << """
+            ${mavenRepository(mavenHttpRepo)}
+            ${customConfigDependencyAssignment(moduleA, moduleB, moduleC)}
+            ${configSyncTask()}
+        """
+
+        when:
+        fails('resolve', '--max-workers=3')
+
+        then:
+        assertDependencyMetaDataReadTimeout(moduleA)
+        assertDependencyMetaDataReadTimeout(moduleB)
+        assertDependencyMetaDataReadTimeout(moduleC)
+        !downloadedLibsDir.isDirectory()
+        where:
+        _ | protocol
+        _ | 'http'
+        _ | 'https'
+    }
+
+    def "skips subsequent dependency resolution if HTTP connection exceeds timeout"() {
+        given:
+        MavenHttpModule moduleB = publishMavenModule(mavenHttpRepo, 'b')
+        MavenHttpModule moduleC = publishMavenModule(mavenHttpRepo, 'c')
+
+        buildFile << """
+            ${mavenRepository(mavenHttpRepo)}
+            ${customConfigDependencyAssignment(moduleA, moduleB, moduleC)}
+            ${configSyncTask()}
+        """
+
+        when:
+        moduleA.pom.expectGetBlocking()
+        fails('resolve', '--max-workers=1')
+
+        then:
+        assertDependencyMetaDataReadTimeout(moduleA)
+        assertDependencySkipped(moduleB)
+        assertDependencySkipped(moduleC)
+        !downloadedLibsDir.isDirectory()
+    }
+
+    def "skipped repositories are only recorded for the time of a single build execution"() {
+        given:
+        MavenHttpModule moduleB = publishMavenModule(mavenHttpRepo, 'b')
+
+        buildFile << """
+            ${mavenRepository(mavenHttpRepo)}
+            ${customConfigDependencyAssignment(moduleA, moduleB)}
+            ${configSyncTask()}
+        """
+
+        when:
+        moduleA.pom.expectGetBlocking()
+        fails('resolve', '--max-workers=1')
+
+        then:
+        assertDependencyMetaDataReadTimeout(moduleA)
+        assertDependencySkipped(moduleB)
+        !downloadedLibsDir.isDirectory()
+
+        when:
+        moduleA.pom.expectGet()
+        moduleA.artifact.expectGet()
+        moduleB.pom.expectGet()
+        moduleB.artifact.expectGet()
+        succeeds('resolve', '--max-workers=1')
+
+        then:
+        downloadedLibsDir.assertContainsDescendants('a-1.0.jar', 'b-1.0.jar')
+    }
+
+    def "fails build if HTTP connection exceeds timeout when resolving metadata"() {
```

Comment:
:x: We should try to unify the logic of these test cases. They are very similar. Maybe break out some fixture methods.

Reply:
Thanks for advice, I've compressed them, would you like to take a look?